lib.nix

```diff
@@ -317,6 +322,11 @@ let
       ${lib.concatMapStrings (p: ''
         ln -s "${p}" "$out/cache/${p.cacheFilename}"
       '') fetchedPackages}
+
+      mkdir $out/index
+      for i in $(seq 0 255); do
+        mkdir $out/index/$(printf "%02x" "$i")
+      done
```

Comment:
Yarn 4 tries to construct this structure in the global folder which fails since it will be readonly so we construct it ourselves.